### PR TITLE
planner: add new variables for instance plan cache

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: ref #54057

Problem Summary: planner: add new variables for instance plan cache

### What changed and how does it work?

Add three new variables for the instance level plan cache:
1. `tidb_enable_instance_plan_cache`: whether to enable this feature;
2. `tidb_instance_plan_cache_target_mem_size`: the target memory limit (or soft mem limit) of the cache;
3. `tidb_instance_plan_cache_max_mem_size`: the max memory limit (or hard mem limit) of the cache;

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by time-and-fate — APPROVED** [review]

File: pkg/sessionctx/variable/sysvar.go [hunk 1]
```diff
@@ -1336,6 +1337,44 @@ var defaultSysVars = []*SysVar{
...
+		SetGlobal: func(_ context.Context, s *SessionVars, val string) error {
+			EnableInstancePlanCache.Store(TiDBOptOn(val))
+			return nil
+		}},
+	{Scope: ScopeGlobal, Name: TiDBInstancePlanCacheTargetMemSize, Value: strconv.Itoa(int(size.MB * 100)), Type: TypeInt, MinValue: 1, MaxValue: math.MaxInt32,
+		GetGlobal: func(_ context.Context, s *SessionVars) (string, error) {
+			return strconv.FormatInt(InstancePlanCacheTargetMemSize.Load(), 10), nil
+		},
+		SetGlobal: func(_ context.Context, s *SessionVars, val string) error {
+			v, err := strconv.ParseInt(val, 10, 64)
```
Comment (zimulala) [inline pkg/sessionctx/variable/sysvar.go:1353]: Do we need to consider using `TidbOptInt64`?  I see that many variables are directly set to the default value when the error occurs, and I wonder whether these two variables are considered to be treated the same.

Comment (qw4990) [reply pkg/sessionctx/variable/sysvar.go]: Updated. PTAL

**Review by zimulala — APPROVED** [review]
LGTM

**Pull request merged**